apps/shared/interval.h

```diff
@@ -18,6 +18,7 @@ class Interval {
   void setEnd(double f);
   void setStep(double f);
   void setElement(int i, double f);
+  void setNeedCompute(bool compute){ m_needCompute = compute;}
```

Comment:
m_needCompute is never set to false through external code, only Interval::computeElements private method will turn m_needCompute to false. So the public API of Interval should only expose a method that can turn m_needCompute to true like `forceComputing()` - you might think of a better name!

Reply:
Ok !